src/libraries/System.Text.Encoding.CodePages/src/System/Text/DecoderNLS.cs

```diff
@@ -87,12 +88,11 @@ public override unsafe int GetCharCount(byte[] bytes, int index, int count, bool
             if (bytes.Length - index < count)
                 throw new ArgumentOutOfRangeException(nameof(bytes), SR.ArgumentOutOfRange_IndexCountBuffer);
 
-            // Avoid null fixed problem
-            if (bytes.Length == 0)
-                bytes = new byte[1];
+            // Avoid empty input fixed problem
+            ref byte b = ref bytes.Length != 0 ? ref bytes[0] : ref Unsafe.AsRef<byte>((void*)1);
```

Comment:
You can use `MemoryMarshal.GetArrayDataReference` instead. It also works with empty arrays and is a JIT intrinsic so should be generally better.

Reply:
This library has net4.6.2 and netstandard2.0 build, and GetArrayDataReference isn't available there.  You think it's worth ifdef'ing it?

Reply:
The IL will be simpler and easier for the runtime to reason about, so if this is perf sensitive it might be worth it.

That being said, for this particular case we're basically pinning and doing some work for `Length == 0`, which will presumably just be returning a char count of `0`, right?

So if we're going to check, couldn't we just fast track that path rather than dealing with the `ref`?

Reply:
> which will presumably just be returning a char count of 0, right?

Some of the cases already have a fast path for whether the input is 0 or not, returning immediately if it's not.  For the rest, I'm assuming they don't for a good reason, and I'm hesitant to try to add a fast path that might impact behavior around exceptions and the like.